fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -1047,6 +1239,15 @@ ACTOR Future<Void> dataDistributionTracker(Reference<InitialDataDistribution> in
 				self.sizeChanges.add(fetchShardMetricsList(&self, req));
 			}
 			when(wait(self.sizeChanges.getResult())) {}
+
+			when(TenantCacheTenantCreated newTenant = waitNext(tenantCreationSignal.getFuture())) {
+				TraceEvent(SevInfo, "TenantCacheTenantCreated")
+				    .detail("Begin", newTenant.keys.begin)
+				    .detail("End", newTenant.keys.end);
+				self.sizeChanges.add(tenantShardSplitter(&self, newTenant.keys));
+				newTenant.reply.send(true);
```

Comment:
Is there any circumstance we reply `false`? I remember the design doc said the tenant creation is a synchronous process so we need to wait for the boundaries being aligned with the tenant prefix then respond success. It follows that we need to send the signal in `tenantShardSplitter`.

(Not sure whether you change the design afterwards

Reply:
Right now, it is not synchronous with the creation of the tenant. This happens in the background, when DD detects a new tenant. I am planing to make the change to make it synchronous at a later point. This is a simpler mechanism that just opens up further development of TADD that depends on the splitting functionality. At this point, there is no reason for this to return false. The goal here is for me to be able to return true or false and let the cache keep track of whether the signal is processed or whether it needs to be retried. If there are any error cases that occur that warrant a retry, this will be false. At this point, in those cases, if any, we just let it go. That is a gap that needs to be filled.